lib/WWW/WebKit2.pm

```diff
@@ -343,6 +348,22 @@ sub init_webkit {
         $self->load_status($load_event);
     });
 
+    $self->view->signal_connect('decide-policy' => sub {
+        my ($view, $decision, $type) = @_;
+        if ($type eq 'navigation-action') {
+            my $action = $decision->get_navigation_action;
+            die "Already running a navigation action to " .
+                $self->active_navigation_action
+                . " when requested "
+                . $action->get_navigation_type . ' ' . $action->get_request->get_uri
+                if $self->active_navigation_action and not $self->navigation_action->is_redirect;
```

Comment:
Is `$self->navigation_action` a typo should it be `$action->navigation_action`?

`$self->navigation_action` is never set.